source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -56,19 +54,41 @@ void CacheFilter::onDestroy() {
   if (lookup_ != nullptr) {
     lookup_->onDestroy();
   }
-  if (insert_queue_ != nullptr) {
-    // The filter can complete and be destroyed while there is still data being
-    // written to the cache. In this case the filter hands ownership of the
-    // queue to itself, which cancels all the callbacks to the filter, but allows
-    // the queue to complete any write operations before deleting itself.
-    //
-    // In the case that the queue is already empty, or in a state which cannot
-    // complete, setSelfOwned will provoke the queue to abort the write operation.
-    insert_queue_->setSelfOwned(std::move(insert_queue_));
-    insert_queue_.reset();
+  if (upstream_request_ != nullptr) {
+    upstream_request_->disconnectFilter();
+    upstream_request_ = nullptr;
   }
 }
 
+void CacheFilter::sendUpstreamRequest(Http::RequestHeaderMap& request_headers) {
+  Router::RouteConstSharedPtr route = decoder_callbacks_->route();
+  const Router::RouteEntry* route_entry = (route == nullptr) ? nullptr : route->routeEntry();
+  if (route_entry == nullptr) {
+    return sendNoRouteResponse();
+  }
+  const auto thread_local_cluster =
```

Comment:
prefer not using 'auto' per style guide. In particular as a reader I don't know if this is an object, ref, shared_ptr,  unique_ptr, or raw ptr.

Reply:
Done.